gadcp/madcp.py

```diff
@@ -844,6 +844,236 @@ def _regrid_amp(self, ens, method="linear"):
             )
         ens.amp_grid = amp_grid
 
+    def _binmap_one_beam(self, ens, beam_number):
+        """Binmap single ping data for a single beam by linear interpolation.
+
+        Mapping is applied to velocity, amplitude and correlation.
+
+        Currently only works for 4 beam ADCP.
+
+        Parameters
+        ----------
+        ens : Bunch
+            An ADCP dataset read by Multiread.
+        beam_number : int
+            An integer from 1 to 4 representing the beam number.
+
+        Returns
+        -------
+        veli : ndarray
+            Mapped velocity.
+        ampi : ndarray
+            Mapped amplitude.
+        cori : ndarray
+            Mapped correlation.
+
+        """
+
+        if beam_number not in [1, 2, 3, 4]:
```

Comment:
Beam references are zero-based in pycurrents, maybe good to stick with this convention?

Reply:
When I load data with pycurrents I get variables like 'vel1', 'vel2', 'vel3', 'vel4', which are consistent with the way that TRDI labels their beams. Where do they use a different convention?

Reply:
Multiread expects the reference for a bad beam as zero-based so I guess it's not all consistent in pycurrents: https://github.com/gunnarvoet/gadcp/blob/main/gadcp/madcp.py#L300

I think we should keep things consistent here. Maybe we open an issue to change the bad beam reference in this package to 1-4? I am happy either way.

Reply:
I think it is more user friendly for the user-facing functions to be consistent with the way TRDI defines things, and then deal with indexing issues internally.

Reply:
I'll raise an issue.